[CS2113-W10-2] ExchangeCourseMapper

chewycj opened the pull request:
ExchangeCourseMapper allows users to plan their course mapping by listing the universities of interest, along with the specific courses and subject codes offered by each school. Users can quickly filter by NUS-coded modules or by partner universities (PU) when they want to view the relevant options.

okkhoy commented on src/main/java/seedu/exchangecoursemapper/command/AddCoursesCommand.java:
```diff
@@ -0,0 +1,170 @@
+    private static final Logger logger = Logger.getLogger(AddCoursesCommand.class.getName());
+
+
+    private static boolean isValidCourseMapping(String nusCourseInput, String puCourseInput,
```
In general, all non-trivial methods need javadoc header comments

okkhoy commented on src/main/java/seedu/exchangecoursemapper/command/AddCoursesCommand.java:
```diff
@@ -0,0 +1,170 @@
+        if (matchPu != null) {
+            courses = jsonObject.getJsonObject(matchPu).getJsonArray(COURSES_ARRAY_LABEL);
+        } else {
+            System.out.println(INVALID_UNIVERSITY_INPUT);
+            System.out.println(LINE_SEPARATOR);
+            System.out.println(LIST_RELEVANT_PU);
+            System.out.println(LINE_SEPARATOR);
```
Can you not use UI to do this job, rather than printing directly?

okkhoy commented on src/main/java/seedu/exchangecoursemapper/command/HelpCommand.java:
```diff
@@ -0,0 +1,89 @@
+        String command = input.replaceFirst("help", "").trim().toLowerCase();
+
+        switch (command) {
+        case "set":
```
need explicit fallthrough comment

okkhoy reviewed (commented):
Overall, the code is clean. Some minor issues are highlighted in the review. Please take a look and fix.

RCPilot1604 commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+
+### Class Diagrams
+Command Package:
+![Class diagram for Commands](images/CommandClass.png)
```
Would it better to split this diagram into smaller, more bite-sized chunks for better readability?

chewycj replied:
Agreed, thank you for the feedback, we will work on it!

RCPilot1604 commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML:
+![Filter Courses Sequence Diagram](images/FilterCoursesCommand.png)
```
Should there be a third activation bar for "print PU and PU course code"?

RCPilot1604 replied:
Also should "throw new IOException()" be depicted as a solid black arrow within Command?

RCPilot1604 commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+* Assertions and logging are used for error handling.
+
+#### Sequence on PlantUML:
+![ListUniCourseCommand sequence diagram](images/ListUniCoursesCommand.png)
```
Should there be an activation bar and dotted return arrow for "print course details"?

MatchaRRR commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+  displaying either the university’s email address or phone number.
+* There are also assertions and logging in place for error handling.
+
+#### Why it is implemented that way:
+* The `execute` method is essential and unique to every command class so inheritance was used.
+* Every method in the class remains maintainable and has one responsibility this allows easy debugging and
+  refactoring.
+* By using inheritance, new command classes can easily extend the functionality of existing ones 
+  which reducing redundancy in the code
+* Logging and assertions helps the team of developers to follow through the command execution.
+
```
I guess this part is redundant and duplicated?

TrungBui32 commented on docs/diagrams/ListSchoolsCommand.puml:
No group (i.e ListSchoolCommand)

TrungBui32 commented on docs/diagrams/FilterCoursesCommand.puml:
```diff
@@ -0,0 +1,41 @@
+FilterCoursesCommand -> Command: createJsonObject()
+activate Command
+
+opt inputStream is null
+    Command -> Command: throw new IOException()
+end
```
Exception arrow should go out

TrungBui32 commented on docs/diagrams/ListSchoolsCommand.puml:
```diff
@@ -0,0 +1,34 @@
+ListSchoolCommand -> Command: createJsonObject()
+activate Command
+
+opt inputStream is null
+    Command -> Command: throw new IOException()
+end
```
Exception arrow should go out

RCPilot1604 reviewed (commented):
Would it be possible extract out repeated sequential diagram segments using reference frames? For instance, ListSchoolCommand, FilterCoursesCommand, ObtainCoursesCommand etc all share createJsonObject which can be extracted out and the differences be depicted in a ref frame. 

Overall, the sequence diagrams are a bit too long and complicated. Splitting them up using ref frames would be a good idea. 

Also would it be a good idea to show the initial set up seqence as well in a sequence diagram that shows the full run-through of the code? Ref frames can be used here as well to simplify the sequence diagrams.

MatchaRRR commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+![Sequence Diagram](images/MiniCommandClass.png)
+
+**The sections below give more details of each component.**
+
+### Class Diagrams
+Command Package:
+![Class diagram for Commands](images/CommandClass.png)
+
+{TODO: Object Diagram}
```
Perhaps you can try to omit some unnecessary details to make it more readable.

MatchaRRR commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+The `UI`, `Parser` and `Storage` components (also shown in the diagram above),
+* defines its API in a class with the same name as the Package
+
+The `Command` component,
```
maybe you can try to illustrate other components here ?

TrungBui32 commented on docs/diagrams/ListSchoolsCommand.puml:
```diff
@@ -0,0 +1,34 @@
+ListSchoolCommand --> ListSchoolCommand
+deactivate ListSchoolCommand
+
+ListSchoolCommand --> CEG_Student
+deactivate ListSchoolCommand
```
Should return something

TrungBui32 commented on docs/diagrams/FilterCoursesCommand.puml:
No group

chewycj replied:
Thank you for the feedback! May we clarify the meaning of "No group"?

MatchaRRR commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+and South-East Asian universities. This command hence helps the users to keep track of their course mapping process.
+
+#### How the feature is implemented:
+* The `AddCoursesCommand` class extends `Command` class where it overrides the `execute` method for
+  custom behaviour.
+* The command first reads a JSON file to obtain the names via `createJsonObject()` method from the
+  superclass.
+* The `trimString` method then removes the `add` command and checks whether the user gave any input after the command. 
+  The method would return the user's input without the command.
+* This input is then passed into the `parseAddCommand()` method to obtain the relevant information: NUS course code, 
+  name of partner university and partner university course code.
+* Along with the JSON Object created from the `createJSONObject()` method, the information extracted from the 
+ `parseAddCommand()` method would be passed to the `isValidInput()` method to verify the user's course mapping.
+* In the `isValidInput()` method, the `getPUCourseList()` method is called to verify the user's partner university is 
+  included in the dataset. An exception is thrown if the university is not found in the dataset.
+* Afterward, the `isValidCourseMapping` checks whether the NUS course code and PU course code are compatible for 
+  course mapping. 
+* If both checks above are passed, the course mapping would be added to the `myList.json` file.
+* Throughout the code, exceptions, assertions and logging are in place for better error handling. 
+* Line Separator is used to ensure readability and ease of use for users.
```
Perhaps adding some diagrams here will make it easier to understand?

limkongkiat commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML
+![Add Courses Sequence Diagram](images/AddCoursesCommand.png)
```
Perhaps you could standardize whether you use User or CEG_Student?

chewycj replied:
Agreed, our team will work on it!

limkongkiat commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML
+![Delete Courses Sequence Diagram](images/DeleteCoursesCommand.png)
```
Perhaps it would be better to standardize whether or not there are activation bars?

limkongkiat commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML
+![Add Courses Sequence Diagram](images/AddCoursesCommand.png)
```
Maybe you could ensure that all arrows are properly connected to the bar?

limkongkiat commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,285 @@
+  readability.
+
+#### Sequence Diagram on PlantUML:
+![Filter Courses Sequence Diagram](images/ObtainContactsCommand.png)
```
Perhaps you could include an [else] label for the first alt block for greater clarity?

limkongkiat reviewed (commented):
Good job overall, just a few nits with some of the diagrams. Keep it up!

glenda-1506 commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,398 @@
+  has been given by the user, the list index is passed into the storage object's `deleteCourse` method to delete the 
+  plan stored at that index.
+* Lastly, the UI object's `printDeleteMessage` is called to inform the user of the course plan which is deleted.
+* Throughout the code, exceptions, assertions and logging are in place for better error handling.
+* Line Separator is used to ensure readability and ease of use for users.
```
As these 2 lines are repeated throughout the DG, perhaps you could put it at the top and just mention it once?

chewycj replied:
That's true, it is repeated throughout our DG 😮  we will extract it out and mention it once, thank you for the feedback!

glenda-1506 commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,398 @@
+  readability.
+
+#### Sequence Diagram on PlantUML:
+![Filter Courses Sequence Diagram](images/ObtainContactsCommand.png)
```
Should this be CEG_Student, instead of User?

glenda-1506 commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,398 @@
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML
+![Delete Courses Sequence Diagram](images/DeleteCoursesCommand.png)
```
Should there be a dotted return arrow for the deleteCourse(listIndex : int)?

glenda-1506 commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,398 @@
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...     | I want to ...                                                   | So that I can ...                                              |
+|---------|--------------|-----------------------------------------------------------------|----------------------------------------------------------------|
+| v1.0    | CEG students | see the possible Oceania and South East Asia partner university | see all my possible choices in those regions                   |
```
I think this is a typo, it should standardise this with the rest?

glenda-1506 reviewed (commented):
Great job! The diagrams were well-drawn with minor errors. Perhaps you could consider colour-coding your sequence diagrams like the Architecture Diagram?

louisjoety committed:
Merge branch 'master' into improvement/parser-test

hzxnancy committed:
Fix checkstyle

hzxnancy committed:
Fix checkstyle

hzxnancy committed:
Remove myList.json from data folder

chewycj committed:
Merge pull request #315 from hzxnancy/branch-UniShorthand

Add method to parse university abbreviations

louisjoety committed:
Extract out the validation segment of obtains

louisjoety committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into improvement/parser-test

louisjoety committed:
Add test cases when abbreviation is used

louisjoety committed:
Restructure obtain contacts file

louisjoety committed:
Enable shortened school name

louisjoety committed:
Add new test cases for shortened school names

chewycj committed:
Fix grammar bugs in delete command messages

chewycj committed:
Add named constants

louisjoety committed:
Update diagrams to new implementation

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-TPD

# Conflicts:
#	src/main/java/seedu/exchangecoursemapper/constants/Messages.java

louisjoety committed:
Make ppp shorter and succint

louisjoety committed:
Shorten length of PPP

chewycj committed:
Modify filter command to take in special NUS course codes

hzxnancy committed:
Disable loggings for AddCoursesCommand, CourseValidator and CourseRepository

chewycj committed:
Modify invalid NUS course code exception message

hzxnancy committed:
Disable loggings for CourseValidator

chewycj committed:
Remove myList.json file

chewycj committed:
Update JUnit tests for filter command

chewycj committed:
Update JUnit tests for delete command

louisjoety committed:
Fix typo around DG implementation

chewycj committed:
Apply standard notations to command class diagrams

hzxnancy committed:
Update UG based on PE-D comments

hzxnancy committed:
Merge pull request #320 from chewycj/branch-bug-fix-command-class-diagram

[Bug Fix] Documentation: Apply standard notations to command class diagrams

louisjoety committed:
Merge branch 'master' into improvement/parser-test

louisjoety committed:
Merge branch 'master' into improvement/ppp-update-2-page

hzxnancy committed:
Merge pull request #319 from chewycj/branch-command-bug-fixes

[Bug Fix] Exception handling in Filter and delete commands

louisjoety committed:
Merge branch 'master' into improvement/parser-test

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-TPD

hzxnancy committed:
Update UG notes section

hzxnancy committed:
Fix checkstyle

hzxnancy committed:
Fix checkstyle

hzxnancy committed:
Remove myList.json to pass test cases

chewycj committed:
Merge pull request #321 from hzxnancy/branch-TPD

Update User Guide and messages from PE-D

louisjoety committed:
Fix merge conflicts

louisjoety committed:
Update UG with better UX format

louisjoety committed:
Remove repeated comments for UG

louisjoety committed:
Update positive test case into UG

louisjoety committed:
Fix checkstyle warnings given

wahkia committed:
Merge pull request #322 from louisjoety/improvement/parser-test

[Improvement] Work on feedback from PED and Refactor Obtain Contacts

Commit:
Refactored UG based on PED

louisjoety committed:
Merge branch 'master' into improvement/ppp-update-2-page

louisjoety committed:
Fix javadocs for SchoolContactValidator class

Commit:
Refactor UI Messages based on PED

Commit:
Fix checkstyle test

Commit:
Fix checkstyle for ListCommnadsCommandTest

chewycj committed:
Remove magic literals from filter and delete commands

chewycj committed:
Remove magic literals from filter and delete commands

chewycj committed:
Merge pull request #323 from wahkia/feature/refactor-UG-PED

Refactor UG and UI based on PED comments

chewycj committed:
Reduce blurriness in class diagrams

chewycj committed:
Reduce blurriness in class diagrams

hzxnancy committed:
Remove NBSP tag from database

hzxnancy committed:
Refactor code

chewycj committed:
Update PPP with newest details

chewycj committed:
Update PPP with newest details

chewycj committed:
Remove negative test cases from UG

chewycj committed:
Remove negative test cases from UG

chewycj committed:
Merge pull request #325 from hzxnancy/branch-databaseBugfix

[BugFix] Remove [NBSP] tag from database

chewycj committed:
Update AboutUs.md with full names

chewycj committed:
Update AboutUs.md with full names

chewycj committed:
Fix documentation bugs in sequence diagrams

chewycj committed:
Fix documentation bugs in sequence diagrams

chewycj committed:
Update documentation for filter command

chewycj committed:
Update documentation for filter command

chewycj committed:
Merge branch 'branch-documentation-bug-fixes' of https://github.com/chewycj/tp into branch-documentation-bug-fixes

louisjoety committed:
Merge pull request #327 from chewycj/branch-documentation-bug-fixes

Chiang-HuiXin committed:
Fix bugs for PE-D

Chiang-HuiXin committed:
Update JUnit test for HelpCommand

Chiang-HuiXin committed:
Modify ListUniCoursesCommand to adopt the abbrevated PU names

hzxnancy committed:
Add javadocs for parser class

Chiang-HuiXin committed:
Change keyword for ListUniCoursesCommand

Change from "set" to "list courses"

chewycj committed:
Fix grammar errors

chewycj committed:
Fix grammar errors

Chiang-HuiXin committed:
Refactor code by extracting code into helper methods

Chiang-HuiXin committed:
Update HelpCommand and Commands for the new "list courses"

Chiang-HuiXin committed:
Update JUnit test for find, list courses and help commands

louisjoety committed:
Merge branch 'master' into improvement/ppp-update-2-page

Chiang-HuiXin committed:
Update DG and UG to replace "set" with "list courses"

louisjoety committed:
Fix UG for Obtain to be clearer

Commit:
Add duplicate entry checker to remove and save unique courses

- Implemented logic to identify and remove duplicate courses from the repository.
- Updated method to save the cleaned course list back to the file.
- Added user notifications for removed duplicates and confirmation of file update.
- Added new validation code to Add,Compare, Delete, Find, List Mapped Commands
- Updated DG for Compare and List Mapped to include new checks

Commit:
Fix Checkstyle unsued imports

louisjoety committed:
Refactor magic variables and logs

louisjoety committed:
Merge pull request #331 from wahkia/feature/add-duplicate-checker

Add duplicate entry checker to remove duplicates and save unique courses

louisjoety committed:
Merge branch 'master' into improvement/ppp-update-2-page

Chiang-HuiXin committed:
Fix checkstyle error in ListCommandsCommandTest

louisjoety committed:
Adjust DG to fit new implementation

louisjoety committed:
Update ppp for better spacing

louisjoety committed:
Fix test cases to avoid repeated testing

louisjoety committed:
Merge pull request #330 from Chiang-HuiXin/branch-Fixbugsv2

[Improvement] Fix bugs for V2.1 and change keyword "set" to "list courses"

louisjoety committed:
Merge branch 'master' into improvement/ppp-update-2-page

louisjoety committed:
Remove mylist things

hzxnancy committed:
Update storage class javadocs

chewycj committed:
Fix grammatical error in DG

chewycj committed:
Fix grammatical error in DG

hzxnancy committed:
Change casing for abbreviations

hzxnancy committed:
Edit test cases to fit lowercase pu

chewycj committed:
Merge pull request #334 from hzxnancy/branch-abbreviationBugFix

[BigFix] Change casing for abbreviations

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-javadocsV2

chewycj committed:
Update DG manual testing with University abbreviations

chewycj committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-documentation-bug-fixes

louisjoety committed:
Resolve merge conflicts

louisjoety committed:
Fix failed test cases

louisjoety committed:
Typo in test case fix again to pass

louisjoety committed:
Adjutst UG to new implementation

louisjoety committed:
Reduce repeated lines in DG

louisjoety committed:
Update test data for list schools

louisjoety committed:
Update the list schools to adapt to shortened names

louisjoety committed:
Update DG with right diagrams

louisjoety committed:
Update the test cases for list schools

louisjoety committed:
Add clarity to loop

louisjoety committed:
Fix error message due to bug in 2 tags

louisjoety committed:
Fix error message

louisjoety committed:
New fix to obtain after updates

Commit:
Add JUnit test for DataIntegrityChecker

Commit:
Refactored CompareMappedCommand to accept abbreviations

Commit:
Fix Checkstyle prroblems

louisjoety committed:
Merge pull request #338 from wahkia/feature/add-data-Integrity-command-test

[Improvement] Add DataIntegrityCheckerTest and refactor CompareMappedCommand to accept abbreviations

louisjoety committed:
Merge branch 'master' into improvement/ppp-update-2-page

Chiang-HuiXin committed:
Modify sequence diagram to make it simpler

hzxnancy committed:
Add javadocs for FileHandler and Storage classes

louisjoety committed:
Fix checkstyle error

louisjoety committed:
Fix indentation

hzxnancy committed:
Add javadocs for DataIntegrityChecker

chewycj committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-documentation-bug-fixes

hzxnancy committed:
Add javadocs for CourseRepository class

louisjoety committed:
Keep function calls simpler

chewycj committed:
Update PPP to be 2 pages

chewycj committed:
Update PPP to be 2 pages

hzxnancy committed:
Merge pull request #339 from louisjoety/improvement/ppp-update-2-page

[Improvement]Ppp Update, update code to match latest version

hzxnancy committed:
Update javadocs

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-javadocsV2

louisjoety committed:
Fix overflows

hzxnancy committed:
Start non void methods Javadocs with "Returns"

louisjoety committed:
Simplify ppp

chewycj committed:
Update PPP to be 2 pages

chewycj committed:
Update PPP overview section

chewycj committed:
Merge branch 'branch-documentation-bug-fixes' of https://github.com/chewycj/tp into branch-documentation-bug-fixes

Commit:
Added checker to add command to prevent addition of duplicate entries

chewycj committed:
Merge pull request #343 from hzxnancy/branch-javadocsV2

[Improvement] Add Javadocs for storage related classes

Chiang-HuiXin committed:
Update FindCoursesCommand sequence diagram to make it simpler

Commit:
Fix checkstyle

louisjoety committed:
Capitalise L

louisjoety committed:
Merge branch 'master' into improvement/ppp-overflow

wahkia committed:
Merge branch 'master' into feature/add-add-command-check

louisjoety committed:
Merge pull request #344 from chewycj/branch-documentation-bug-fixes

[Improvement] Update documentations and PPP based on new code

louisjoety committed:
Merge branch 'master' into improvement/ppp-overflow

Chiang-HuiXin committed:
Update PPP

chewycj committed:
Merge pull request #345 from wahkia/feature/add-add-command-check

Added checker to add command to prevent addition of duplicate entries

hzxnancy committed:
Merge pull request #347 from Chiang-HuiXin/branch-LastCheck

[Improvement] Update Sequence diagram for ListUniCoursesCommand and FindCoursesCommand

chewycj committed:
Update logs in code

chewycj committed:
Update PPP to be 2 pages

chewycj committed:
Fix merge conflicts in delete command

chewycj committed:
Adjust layout of PPP

hzxnancy committed:
Update AddCoursesCommand seq diagram

Chiang-HuiXin committed:
Update Parser Class diagram

wahkia committed:
Merge pull request #350 from chewycj/branch-clean-up-code

[Improvement] Add loggings to new parts of code, and update PPP

Chiang-HuiXin committed:
Update DG so that the sequence diagram shows

Chiang-HuiXin committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-ParserClassDiagram

louisjoety committed:
Merge branch 'master' into improvement/ppp-overflow

hzxnancy committed:
Update PPP

Chiang-HuiXin committed:
Update DG for FindCoursesCommand

louisjoety committed:
Merge pull request #352 from Chiang-HuiXin/branch-ParserClassDiagram

[Improvement] Update Parser Class diagram for DG

louisjoety committed:
Merge branch 'master' into improvement/ppp-overflow

hzxnancy committed:
Update DG with university abbreviation methods

hzxnancy committed:
Update CourseValidator sequence diagram

louisjoety committed:
Remove trailing spaces

chewycj committed:
Merge pull request #353 from louisjoety/improvement/ppp-overflow

[Improvement] Fix typos and PPP overflow in PDF

Chiang-HuiXin committed:
Shorten PPP

Chiang-HuiXin committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-fixppp

Chiang-HuiXin committed:
Fix UG for FindCommand

hzxnancy committed:
Update UG

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-DocsUpdatev6

hzxnancy committed:
Merge pull request #354 from Chiang-HuiXin/branch-fixppp

[Improvement] Refactor and shorten PPP

chewycj committed:
Merge pull request #356 from hzxnancy/branch-DocsUpdatev6

[Improvement] Update PPP, DG and UG after v2.1

chewycj committed:
Adjust layout of PPP

chewycj committed:
Update JUnit tests for Filter and Delete commands

Chiang-HuiXin committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-LastChecksv2

louisjoety committed:
Merge pull request #357 from chewycj/branch-ppp-v2-1

Fix PPP overflow and JUnit tests

louisjoety committed:
Add colons into DG

Chiang-HuiXin committed:
Shorten PPP

Chiang-HuiXin committed:
Delete FindCoursesCommand sequence diagram image

hzxnancy committed:
Add page breaks for better formatting

Chiang-HuiXin committed:
Add FindCoursesCommand sequence diagram image back

chewycj committed:
Merge pull request #358 from Chiang-HuiXin/branch-LastChecksv2

[Improvement] Shorten PPP and resize FindCoursesCommand sequence diagram

Commit:
Refactored diagrams and added javadocs ListPersonalTrackerCommand nad CompareMappedCommand

louisjoety committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into improvement/typo-fixes

chewycj committed:
Merge pull request #359 from wahkia/feature/refactor-diagrams

Refactored diagrams and added javadocs ListPersonalTrackerCommand and CompareMappedCommand

hzxnancy committed:
Update PPP with more github links

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-finalUpdates

louisjoety committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into improvement/typo-fixes

louisjoety committed:
Fix entry point

hzxnancy committed:
Extract magic numbers and constants

louisjoety committed:
Refactor obtain and list schools

louisjoety committed:
Refactor obtain and list schools

Commit:
Update PPP based on latest changes

hzxnancy committed:
Add javadocs for Course class

hzxnancy committed:
Suppress loggings in storage classes

hzxnancy committed:
Merge pull request #360 from wahkia/feature/update-ppp

Update PPP to meet requirements based on latest changes

hzxnancy committed:
Merge pull request #361 from louisjoety/improvement/typo-fixes

[Improvement] Minor typo fixes

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-finalUpdates

hzxnancy committed:
Fix checkstyle

Chiang-HuiXin committed:
Shorten PPP to adhere to 2 page limit

Chiang-HuiXin committed:
Shorten PPP further

Chiang-HuiXin committed:
Merge pull request #362 from hzxnancy/branch-finalUpdates

[Improvement] Final updates on code

hzxnancy committed:
Merge pull request #363 from Chiang-HuiXin/branch-FixPPPv2

[Improvement] Shorten PPP to adhere to 2 page limit

hzxnancy committed:
Fix hyperlinks for DG and UG

hzxnancy committed:
Fix grammar in UG

louisjoety committed:
Merge pull request #365 from hzxnancy/branch-fixLinks

Fix hyperlinks for DG and UG

hzxnancy committed:
Remove line break from DG

hzxnancy committed:
Update CourseValidator sequence diagram

Commit:
Update PPP to fit page limit

hzxnancy committed:
Edit grammar mistakes

hzxnancy committed:
Remove test case from UG for clairty

hzxnancy committed:
Fix minor grammar mistakes

hzxnancy committed:
Add javadocs for removeDuplicatteCourses()

hzxnancy committed:
Merge pull request #366 from wahkia/feature/refactor-PPP

Update PPP to fit page limit

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-DocsUpdatev7

hzxnancy committed:
Fix minor grammar mistakes in DG

hzxnancy committed:
Update UG example description

hzxnancy committed:
Update test case to follow new message

chewycj committed:
Merge pull request #367 from hzxnancy/branch-DocsUpdatev7

[Improvement] Fix minor grammatical errors and update CourseValidator sequence diagram

chewycj committed:
Update DG manual testing section

chewycj committed:
Update DG instructions for data corruption

chewycj committed:
Update DG instructions

chewycj committed:
Update DG instructions

hzxnancy committed:
Merge pull request #369 from chewycj/branch-dg-bug-fixes

Update DG manual testing section

louisjoety committed:
Add List schools DG implementation causes

louisjoety committed:
Update DG to prevent overflow in checkstyle

louisjoety committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into improvement/List-schools-implementation

louisjoety committed:
Match DG to SD

louisjoety committed:
Update the DG with new obtain diagram rectifications

Chiang-HuiXin committed:
Modify sequence diagram

Chiang-HuiXin committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-FixSequenceDiagram

louisjoety committed:
Fix obtain diagram

Chiang-HuiXin committed:
Update Sequence diagram for the 3 classes

hzxnancy committed:
Extract method to align with seq diagram

chewycj committed:
Merge pull request #371 from Chiang-HuiXin/branch-FixSequenceDiagram

[Improvement] Update Sequence diagrams

chewycj committed:
Merge pull request #370 from louisjoety/improvement/List-schools-implementation

[Improvement] Update List school why it is implemented

hzxnancy committed:
Remove unnecessary lines

hzxnancy committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-2/tp into branch-extractmethod

louisjoety committed:
Switch trivial statement

louisjoety committed:
Update photo on DG

chewycj committed:
Merge pull request #372 from hzxnancy/branch-extractmethod

Extract method to align with AddCoursesCommand seq diagram

chewycj committed:
Merge pull request #373 from louisjoety/improvement/display-imeage

[Improvement] Display Image on DG

Commit:
Refactored ListCommandCommands UG

louisjoety committed:
Merge pull request #374 from wahkia/feature/refactor-list-commands-UG
